apollo-common/src/main/java/com/ctrip/framework/apollo/common/utils/ConfigValidator.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2022 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.apollo.common.utils;
+
+import com.ctrip.framework.apollo.core.utils.StringUtils;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.util.Assert;
+
+/**
+ * Config Validator
+ *
+ * @author Lv Lifeng
+ */
+public class ConfigValidator {
+
+  private final ConfigurableApplicationContext applicationContext;
+
+  public ConfigValidator(ConfigurableApplicationContext applicationContext) {
+    this.applicationContext = applicationContext;
+    checkDataBaseName();
+  }
+
+  private void checkDataBaseName() {
+    String dataSourceUrl = System.getProperty("spring.datasource.url");
+    if (!StringUtils.isBlank(dataSourceUrl)
+        && dataSourceUrl.contains("/")
+        && dataSourceUrl .contains("?")
+        && !StringUtils.isBlank(applicationContext.getId())) {
+      String dbName = dataSourceUrl.substring(dataSourceUrl.lastIndexOf("/"), dataSourceUrl.indexOf("?"));
+      if (!StringUtils.isEmpty(dbName)) {
+        switch (applicationContext.getId()) {
```

Comment:
Why not using `spring.application.name`?

Reply:
Because the value of `spring.application.name` is finally assigned to  `applicationContext.id`,so I used this parameter directly.
The following is springboot source code.`org.springframework.boot.context.ContextIdApplicationContextInitializer`
```java
        @Override
	public void initialize(ConfigurableApplicationContext applicationContext) {
		ContextId contextId = getContextId(applicationContext);
		applicationContext.setId(contextId.getId());
		applicationContext.getBeanFactory().registerSingleton(ContextId.class.getName(), contextId);
	}

	private ContextId getContextId(ConfigurableApplicationContext applicationContext) {
		ApplicationContext parent = applicationContext.getParent();
		if (parent != null && parent.containsBean(ContextId.class.getName())) {
			return parent.getBean(ContextId.class).createChildId();
		}
		return new ContextId(getApplicationId(applicationContext.getEnvironment()));
	}

	private String getApplicationId(ConfigurableEnvironment environment) {
		String name = environment.getProperty("spring.application.name");
		return StringUtils.hasText(name) ? name : "application";
	}
```

Reply:
OK, but I think `environment.getProperty("spring.application.name")` may be more easy for people to understand.

Reply:
I agree completely. :-)